app/Http/Controllers/ActiviteController.php

```diff
@@ -33,32 +42,101 @@ public function index()
 
         //Only the admin can see all the activities. Normal user sees the zone activities.
         if(auth()->user()->isAdmin()){
-            $activites = Activite::query()->orderby('nom', 'asc')->get();
+            $activites = Activite::query()->orderby('nom', 'asc')->with(['categorieActivite']);
         }else{
-            $allActivites = Activite::query()->orderby('nom', 'asc')->get();
-            //Get only zone related activities
-            //if activity of zone see only for user zone
-            //If activity of sous-zone then see only sous-zone related
-            //if activity of group then see only grpup related
-            $activites = [];
-            foreach($allActivites as $activite){
+            // TODO add with to eager load datas and enhance performance 
+            $activites = Activite::query()
+                ->where(function ($query) use ($groupe, $sousZone, $zone) {
+                    $query->where(function ($subQuery) use ($groupe) {
+                        // ACTIVITE_GROUPE
+                        $subQuery->where('type_activite', Constantes::ACTIVITE_GROUPE)
+                                ->where('groupe_id', $groupe->id);
+                    })
+                    ->orWhere(function ($subQuery) use ($sousZone) {
+                        // ACTIVITE_SOUS_ZONALE
+                        $subQuery->where('type_activite', Constantes::ACTIVITE_SOUS_ZONALE)
+                                ->where('sous_zone_id', $sousZone->id);
+                    })
+                    ->orWhere(function ($subQuery) use ($zone) {
+                        // ACTIVITE_ZONALE
+                        $subQuery->where('type_activite', Constantes::ACTIVITE_ZONALE)
+                                ->where('zone_id', $zone->id);
+                    })
```

Comment:
Just for my own knowledge is this the equivalent of 
`select * from activities where (type_activite == 'ACTIVE_GROUPE'  and group_id = group_id) or (type_activite == 'ACTIVITY_SOUS_ZONE'  and sous_zone_id = sous_zone_id) or (type_activite == 'ACTIVITY_ZONE'  and zone_id = zone_id)`

If that is the case, how do we know this is going to produce the expected result, I can see a combination of inputs that can return activities for both sous zones and zones.

can we add some unit tests to validate behaviors?